share/smack/lib/smack.c

```diff
@@ -738,6 +795,30 @@ void __SMACK_decls(void) {
   DECLARE(BUILTIN_RMODE_CONVERSION, bv16, bvdouble, $si2fp,(_ to_fp 11 53));
   DECLARE(BUILTIN_RMODE_CONVERSION, bv8, bvdouble, $si2fp,(_ to_fp 11 53));
   DECLARE(BUILTIN_RMODE_CONVERSION, bv1, bvdouble, $si2fp,(_ to_fp 11 53));
+  DECLARE(BUILTIN_RMODE_CONVERSION, bv128, bvlongdouble, $ui2fp,(_ to_fp_unsigned 11 53));
+  DECLARE(BUILTIN_RMODE_CONVERSION, bv96, bvlongdouble, $ui2fp,(_ to_fp_unsigned 11 53));
+  DECLARE(BUILTIN_RMODE_CONVERSION, bv88, bvlongdouble, $ui2fp,(_ to_fp_unsigned 11 53));
+  DECLARE(BUILTIN_RMODE_CONVERSION, bv64, bvlongdouble, $ui2fp,(_ to_fp_unsigned 11 53));
+  DECLARE(BUILTIN_RMODE_CONVERSION, bv56, bvlongdouble, $ui2fp,(_ to_fp_unsigned 11 53));
+  DECLARE(BUILTIN_RMODE_CONVERSION, bv48, bvlongdouble, $ui2fp,(_ to_fp_unsigned 11 53));
+  DECLARE(BUILTIN_RMODE_CONVERSION, bv40, bvlongdouble, $ui2fp,(_ to_fp_unsigned 11 53));
+  DECLARE(BUILTIN_RMODE_CONVERSION, bv32, bvlongdouble, $ui2fp,(_ to_fp_unsigned 11 53));
+  DECLARE(BUILTIN_RMODE_CONVERSION, bv24, bvlongdouble, $ui2fp,(_ to_fp_unsigned 11 53));
+  DECLARE(BUILTIN_RMODE_CONVERSION, bv16, bvlongdouble, $ui2fp,(_ to_fp_unsigned 11 53));
+  DECLARE(BUILTIN_RMODE_CONVERSION, bv8, bvlongdouble, $ui2fp,(_ to_fp_unsigned 11 53));
+  DECLARE(BUILTIN_RMODE_CONVERSION, bv1, bvlongdouble, $ui2fp,(_ to_fp_unsigned 11 53));
+  DECLARE(BUILTIN_RMODE_CONVERSION, bv128, bvlongdouble, $si2fp,(_ to_fp 11 53));
```

Comment:
Wait, these lines seem strange to me. Should this really be `_ to_fp 11 53`?
Doesn't `long double` have more bits typically?